File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        s = list(s)
        left = 0
        right = len(s) - 1
        vowel = ['a','e','i','o','u']

        while left < right:
            if s[left].lower() in vowel and s[right].lower() in vowel:
                s[left],s[right] = s[right],s[left]
                left += 1
                right -= 1
            
            if s[left].lower() not in vowel:
                left += 1
            
            if s[right].lower() not in vowel:
                right -= 1
        
        return "".join(s)

File: 0345-reverse-vowels-of-a-string/0345-reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(f) {
    let s = f.split('')
    let array = ['a','e','i','o','u'];
    console.log(s)
    let left = 0, right = s.length - 1

    while (left < right) {
        while (left < right && !array.includes(s[left].toLowerCase())) {
            left++
        }
        while (left < right && !array.includes(s[right].toLowerCase())) {
            right--
        }
        [s[left], s[right]] = [s[right], s[left]]
        left++
        right--
    }

    return s.join('')
};